rust/kernel/miscdev.rs

```diff
@@ -73,17 +78,12 @@ impl Registration {
     }
 }
 
-impl Default for Registration {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 // SAFETY: The only method is `register()`, which requires a (pinned) mutable `Registration`, so it
-// is safe to pass `&Registration` to multiple threads because it offers no interior mutability.
-unsafe impl Sync for Registration {}
+// is safe to pass `&Registration` to multiple threads because it offers no interior mutability,
+// except maybe through [`Registration::context`], but it is itself [`Sync`].
+unsafe impl<T: Sync> Sync for Registration<T> {}
 
-impl Drop for Registration {
+impl<T: Sync> Drop for Registration<T> {
```

Comment:
Does `T` need to be `Send` as well so it can be dropped from any thread?

Reply:
No because its ownership remains with its creator, so it can ensure that it's called from the same thread.

I'm making `Registration` `Send` (if `T` is as well) so that the creator can destroy it from any thread if it wants to.